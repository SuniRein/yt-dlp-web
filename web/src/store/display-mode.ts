import { defineStore } from 'pinia';
import { ref } from 'vue';

type displayModeState = 'light' | 'dark';

export const useDisplayModeStore = defineStore('displayMode', () => {
    const state = ref<displayModeState>(window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light');

    function toggle() {
        state.value = state.value === 'light' ? 'dark' : 'light';
    }

    return {
        state,
        toggle,
    };
});
