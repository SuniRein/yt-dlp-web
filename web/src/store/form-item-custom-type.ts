import { reactive } from 'vue';

export interface FormItemCustomType {
    type: string;
    validator: (value: string) => boolean;
    validityMessage: string;
}

interface FormItemCustomTypeMap {
    [key: string]: Omit<FormItemCustomType, 'type'>;
}

export const formItemCustomTypes = reactive<FormItemCustomTypeMap>({});

export function registerCustomType(customType: FormItemCustomType) {
    const { type, ...rest } = customType;
    formItemCustomTypes[type] = rest;
}
