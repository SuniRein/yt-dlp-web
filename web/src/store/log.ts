import { defineStore } from 'pinia';
import { ref } from 'vue';

export const useLogStore = defineStore('log', () => {
    const value = ref('');

    return {
        value,

        clear() {
            value.value = '';
        },

        log(message: string) {
            value.value += `${message}\n`;
        },
    };
});
