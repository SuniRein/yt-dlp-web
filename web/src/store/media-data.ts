import { defineStore } from 'pinia';
import { ref } from 'vue';
import type { MediaData } from '@/types/MediaData.types';

export const useMediaDataStore = defineStore('mediaData', () => {
    const value = ref<MediaData | null>();

    return {
        value,

        clear() {
            value.value = null;
        },
    };
});
