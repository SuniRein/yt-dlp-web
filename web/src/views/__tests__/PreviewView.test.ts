import PreviewView from '@/views/PreviewView.vue';

import { setActivePinia, createPinia } from 'pinia';
import { useMediaDataStore } from '@/store/media-data';
import type { MediaData } from '@/types/MediaData.types';

import { mount } from '@vue/test-utils';
import { test, expect } from 'vitest';

import info from './media-info.json';

setActivePinia(createPinia());
const mediaData = useMediaDataStore();

const preview = mount(PreviewView);

function getContent() {
    return preview.find('[data-test="preview-content"]');
}

function getStaticHtml() {
    return getContent()
        .html()
        .replace(/data-n-id=".*?"/g, '')
        .replace(/data-v-.{8}=".*?"/g, '')
        .replace(/data-test=".*?"/g, '');
}

test('create element', () => {
    expect(getContent().exists()).toBe(false);
});

test('show preview', async () => {
    mediaData.value = info as MediaData;
    await preview.vm.$nextTick();

    expect(getContent().exists()).toBe(true);
    expect(getStaticHtml()).toMatchSnapshot();
});

test('clear preview', async () => {
    const clearButton = preview.get('[data-test="preview-clear-button"]');
    await clearButton.trigger('click');

    expect(getContent().exists()).toBe(false);
    expect(mediaData.value).toBeNull();
});
