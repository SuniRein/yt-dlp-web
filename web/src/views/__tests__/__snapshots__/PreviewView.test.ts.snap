// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`show preview 1`] = `
"<div >
  <div class="n-card n-card--bordered" style="--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 3px; --n-color: #fff; --n-color-modal: #fff; --n-color-popover: #fff; --n-color-embedded: rgb(250, 250, 252); --n-color-embedded-modal: rgb(250, 250, 252); --n-color-embedded-popover: rgb(250, 250, 252); --n-color-target: #18a058; --n-text-color: rgb(51, 54, 57); --n-line-height: 1.6; --n-action-color: rgb(250, 250, 252); --n-title-text-color: rgb(31, 34, 37); --n-title-font-weight: 500; --n-close-icon-color: rgba(102, 102, 102, 1); --n-close-icon-color-hover: rgba(102, 102, 102, 1); --n-close-icon-color-pressed: rgba(102, 102, 102, 1); --n-close-color-hover: rgba(0, 0, 0, .09); --n-close-color-pressed: rgba(0, 0, 0, .13); --n-border-color: rgb(239, 239, 245); --n-box-shadow: 0 1px 2px -2px rgba(0, 0, 0, .08), 0 3px 6px 0 rgba(0, 0, 0, .06), 0 5px 12px 4px rgba(0, 0, 0, .04); --n-padding-top: 19px; --n-padding-bottom: 20px; --n-padding-left: 24px; --n-font-size: 14px; --n-title-font-size: 18px; --n-close-size: 22px; --n-close-icon-size: 18px; --n-close-border-radius: 3px;">
    <!---->
    <div class="n-card-header" role="heading">
      <div class="n-card-header__main" role="heading">RLCraft 最新Dregora版生存 EP5</div>
      <!---->
      <!---->
    </div>
    <div class="n-card__content" role="none">点赞过3万更新新的一集~

      EP1：BV17kyaY6E46
      EP2：BV1cvDAY5EKj
      EP3：BV1SMSmY2Ew1
      EP4：BV1TFDsYUE2M <div role="none" class="n-image"><img referrerpolicy="no-referrer" src="http://i1.hdslb.com/bfs/archive/c0ac572f42668c212cf9c6a34fbe4a19aa0397f4.jpg" loading="eager" style="object-fit: fill;" data-error="false" data-preview-src="http://i1.hdslb.com/bfs/archive/c0ac572f42668c212cf9c6a34fbe4a19aa0397f4.jpg" data-group-id="">
        <!---->
        <!---->
      </div>
      <p><b>Destination:</b> RLCraft 最新Dregora版生存 EP5 [BV122UxY7Ev4].mp4</p>
    </div>
    <!---->
    <!---->
  </div>
  <div class="n-config-provider">
    <div class="n-data-table n-data-table--bordered n-data-table--single-line" style="--n-font-size: 14px; --n-th-padding: 12px; --n-td-padding: 12px; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 3px; --n-line-height: 1.6; --n-border-color: rgba(239, 239, 245, 1); --n-border-color-modal: rgba(239, 239, 245, 1); --n-border-color-popover: rgba(239, 239, 245, 1); --n-th-color: rgba(250, 250, 252, 1); --n-th-color-hover: rgba(243, 243, 247, 1); --n-th-color-modal: rgba(250, 250, 252, 1); --n-th-color-hover-modal: rgba(243, 243, 247, 1); --n-th-color-popover: rgba(250, 250, 252, 1); --n-th-color-hover-popover: rgba(243, 243, 247, 1); --n-td-color: #fff; --n-td-color-hover: rgba(247, 247, 250, 1); --n-td-color-modal: #fff; --n-td-color-hover-modal: rgba(247, 247, 250, 1); --n-td-color-popover: #fff; --n-td-color-hover-popover: rgba(247, 247, 250, 1); --n-th-text-color: rgb(31, 34, 37); --n-td-text-color: rgb(51, 54, 57); --n-th-font-weight: 500; --n-th-button-color-hover: rgba(0, 0, 100, 0.03); --n-th-icon-color: rgba(194, 194, 194, 1); --n-th-icon-color-active: #18a058; --n-filter-size: 15px; --n-pagination-margin: 12px 0 0 0; --n-empty-padding: 48px 0; --n-box-shadow-before: inset -12px 0 8px -12px rgba(0, 0, 0, .18); --n-box-shadow-after: inset 12px 0 8px -12px rgba(0, 0, 0, .18); --n-sorter-size: 15px; --n-resizable-container-size: 8px; --n-resizable-size: 2px; --n-loading-size: 28px; --n-loading-color: #18a058; --n-opacity-loading: 0.5; --n-td-color-striped: rgba(250, 250, 252, 1); --n-td-color-striped-modal: rgba(250, 250, 252, 1); --n-td-color-striped-popover: rgba(250, 250, 252, 1);">
      <div class="n-data-table-wrapper">
        <div class="n-data-table-base-table">
          <!---->
          <div class="n-data-table-base-table-body n-scrollbar" style="--n-scrollbar-bezier: cubic-bezier(.4, 0, .2, 1); --n-scrollbar-color: rgba(0, 0, 0, 0.25); --n-scrollbar-color-hover: rgba(0, 0, 0, 0.4); --n-scrollbar-border-radius: 5px; --n-scrollbar-width: 5px; --n-scrollbar-height: 5px; --n-scrollbar-rail-inset-horizontal-top: 4px 2px auto 2px; --n-scrollbar-rail-inset-horizontal-bottom: auto 2px 4px 2px; --n-scrollbar-rail-inset-vertical-right: 2px 4px 2px auto; --n-scrollbar-rail-inset-vertical-left: 2px auto 2px 4px; --n-scrollbar-rail-color: transparent;" role="none">
            <div role="none" class="n-scrollbar-container">
              <div role="none" style="min-width: 100%;" class="n-scrollbar-content">
                <table class="n-data-table-table" style="table-layout: auto;">
                  <colgroup>
                    <col>
                    <col>
                    <col>
                    <col>
                    <col>
                    <col>
                    <col>
                    <col>
                    <col>
                    <col>
                    <col>
                    <col>
                  </colgroup>
                  <thead class="n-data-table-thead" >
                    <tr class="n-data-table-tr">
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">ID</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">Extension</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">Resolution</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">FPS</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">Size</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">TBR</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">Protocol</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">VCodec</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">VBR</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">ACodec</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-th">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">ABR</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                      <th style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-th n-data-table-th--last">
                        <div class="n-data-table-th__title-wrapper">
                          <div class="n-data-table-th__title">Chosen</div>
                          <!---->
                        </div>
                        <!---->
                        <!---->
                      </th>
                    </tr>
                  </thead>
                  <tbody  class="n-data-table-tbody">
                    <tr class="n-data-table-tr">
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-td n-data-table-td--center-align">
                        <!---->30216
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-td n-data-table-td--center-align">
                        <!---->m4a
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-td n-data-table-td--center-align">
                        <!---->audio only
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-td n-data-table-td--center-align">
                        <!---->8.63 MB
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->49.111
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-td n-data-table-td--center-align">
                        <!---->https
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->mp4a.40.5
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->49.111
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-col">
                        <!---->
                      </td>
                    </tr>
                    <tr class="n-data-table-tr">
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-td n-data-table-td--center-align">
                        <!---->30232
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-td n-data-table-td--center-align">
                        <!---->m4a
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-td n-data-table-td--center-align">
                        <!---->audio only
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-td n-data-table-td--center-align">
                        <!---->18.35 MB
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->104.386
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-td n-data-table-td--center-align">
                        <!---->https
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->mp4a.40.2
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->104.386
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-col">
                        <!---->
                      </td>
                    </tr>
                    <tr class="n-data-table-tr">
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-td n-data-table-td--center-align">
                        <!---->30280
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-td n-data-table-td--center-align">
                        <!---->m4a
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-td n-data-table-td--center-align">
                        <!---->audio only
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-td n-data-table-td--center-align">
                        <!---->34.80 MB
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->198.012
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-td n-data-table-td--center-align">
                        <!---->https
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->mp4a.40.2
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->198.012
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-col">
                        <!---->Yes
                      </td>
                    </tr>
                    <tr class="n-data-table-tr">
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-td n-data-table-td--center-align">
                        <!---->100046
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-td n-data-table-td--center-align">
                        <!---->mp4
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-td n-data-table-td--center-align">
                        <!---->640x360
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-td n-data-table-td--center-align">
                        <!---->30
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-td n-data-table-td--center-align">
                        <!---->26.17 MB
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->148.881
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-td n-data-table-td--center-align">
                        <!---->https
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->avc1.64001E
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->148.881
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-col">
                        <!---->
                      </td>
                    </tr>
                    <tr class="n-data-table-tr">
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-td n-data-table-td--center-align">
                        <!---->100109
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-td n-data-table-td--center-align">
                        <!---->mp4
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-td n-data-table-td--center-align">
                        <!---->640x360
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-td n-data-table-td--center-align">
                        <!---->30
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-td n-data-table-td--center-align">
                        <!---->28.73 MB
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->163.436
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-td n-data-table-td--center-align">
                        <!---->https
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->hev1.1.6.L120.90
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->163.436
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-col">
                        <!---->
                      </td>
                    </tr>
                    <tr class="n-data-table-tr">
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-td n-data-table-td--center-align">
                        <!---->100022
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-td n-data-table-td--center-align">
                        <!---->mp4
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-td n-data-table-td--center-align">
                        <!---->640x360
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-td n-data-table-td--center-align">
                        <!---->30
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-td n-data-table-td--center-align">
                        <!---->25.19 MB
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->143.347
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-td n-data-table-td--center-align">
                        <!---->https
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->av01.0.00M.10.0.110.01.01.01.0
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->143.347
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-col">
                        <!---->
                      </td>
                    </tr>
                    <tr class="n-data-table-tr">
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-td n-data-table-td--center-align">
                        <!---->100047
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-td n-data-table-td--center-align">
                        <!---->mp4
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-td n-data-table-td--center-align">
                        <!---->852x480
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-td n-data-table-td--center-align">
                        <!---->30
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-td n-data-table-td--center-align">
                        <!---->48.32 MB
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->274.904
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-td n-data-table-td--center-align">
                        <!---->https
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->avc1.64001F
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->274.904
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-col">
                        <!---->
                      </td>
                    </tr>
                    <tr class="n-data-table-tr">
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-td n-data-table-td--center-align">
                        <!---->100110
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-td n-data-table-td--center-align">
                        <!---->mp4
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-td n-data-table-td--center-align">
                        <!---->852x480
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-td n-data-table-td--center-align">
                        <!---->30
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-td n-data-table-td--center-align">
                        <!---->42.48 MB
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->241.675
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-td n-data-table-td--center-align">
                        <!---->https
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->hev1.1.6.L120.90
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->241.675
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-td n-data-table-td--center-align">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-col">
                        <!---->
                      </td>
                    </tr>
                    <tr class="n-data-table-tr">
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ID" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->100023
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Extension" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->mp4
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Resolution" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->852x480
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="FPS" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->30
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Size" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->37.57 MB
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="TBR" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->213.744
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Protocol" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->https
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VCodec" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->av01.0.00M.10.0.110.01.01.01.0
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="VBR" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->213.744
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ACodec" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="ABR" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-row">
                        <!---->
                      </td>
                      <td style="text-align: center;" colspan="1" rowspan="1" data-col-key="Chosen" class="n-data-table-td n-data-table-td--center-align n-data-table-td--last-col n-data-table-td--last-row">
                        <!---->Yes
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div class="n-scrollbar-rail n-scrollbar-rail--vertical n-scrollbar-rail--vertical--right n-scrollbar-rail--disabled" data-scrollbar-rail="true" style="z-index: 3;" aria-hidden="true">
              <transition-stub name="fade-in-transition" appear="false" persisted="false" css="true">
                <!---->
              </transition-stub>
            </div>
            <div class="n-scrollbar-rail n-scrollbar-rail--horizontal n-scrollbar-rail--horizontal--bottom n-scrollbar-rail--disabled" style="z-index: 3;" data-scrollbar-rail="true" aria-hidden="true">
              <transition-stub name="fade-in-transition" appear="false" persisted="false" css="true">
                <!---->
              </transition-stub>
            </div>
          </div>
        </div>
      </div>
      <!---->
      <transition-stub name="fade-in-scale-up-transition" appear="false" persisted="false" css="true">
        <!---->
      </transition-stub>
    </div>
  </div>
</div>"
`;
